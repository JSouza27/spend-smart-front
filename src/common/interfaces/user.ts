export interface IUser {
  id: string;
  name: string;
  email: string;
  imageUrl: null;
  additional_user_id: null;
  account: string;
  nick: string;
}
