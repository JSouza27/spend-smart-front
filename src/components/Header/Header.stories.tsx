import { ComponentMeta, Story } from '@storybook/react';
import { ThemeProvider } from 'styled-components';

import Header from '.';
import theme from '../../styles/theme';
import mockUsuario from '../../data/mocks/mockUsuario';

export default {
  title: 'Header',
  component: Header
} as ComponentMeta<typeof Header>;

export const Basic: Story = (args) => (
  <ThemeProvider theme={theme}>
    <Header user={mockUsuario} />
  </ThemeProvider>
);
