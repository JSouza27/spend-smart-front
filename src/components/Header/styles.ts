import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.neutral_100};
    border: none;
    border-radius: ${theme.border.radius};
    display: flex;
    justify-content: flex-end;
    padding: 1.6rem;
    width: 100%;

    .salutation {
      display: none;
    }

    @media (min-width: 767px) {
      justify-content: space-between;

      .salutation {
        display: block;
      }
    }
  `}
`;
