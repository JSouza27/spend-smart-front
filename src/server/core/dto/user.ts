export interface IUser {
  id: string;
  name: string;
  email: string;
  imageUrl: string | null;
  additional_user_id: string | null;
  cpf?: string;
  telefone?: string;
  account: string;
  nick: string;
}
