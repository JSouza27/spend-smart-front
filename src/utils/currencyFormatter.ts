export class CurrencyFormatter {
  private static language = 'pt-BR';
  private static currency = 'BRL';

  static formatter(num: number): string {
    return (num ?? 0).toLocaleString(this.language, {
      currency: this.currency
    });
  }

  static unformat(num: string): number {
    const nums = num.replace(/[^0-9]+/g, '');
    const i = nums.length - 2;

    return Number(`${nums.substring(0, i)}.${nums.substring(i)}`);
  }
}
