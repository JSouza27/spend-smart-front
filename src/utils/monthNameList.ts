export const monthNameList = [
  'janeiro',
  'fevereiro',
  'mar√ßo',
  'abril',
  'maio',
  'junho',
  'julho',
  'agosto',
  'setembro',
  'outubro',
  'novembro',
  'dezembro'
];
