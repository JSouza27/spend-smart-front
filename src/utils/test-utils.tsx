import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../styles/theme';
import { FormProvider, useForm } from 'react-hook-form';

const AllTheProviders = ({ children }: any) => {
  const methods = useForm();

  return (
    <ThemeProvider theme={theme}>
      <FormProvider {...methods}>{children}</FormProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: any
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
