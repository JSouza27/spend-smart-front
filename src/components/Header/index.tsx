import Heading from '../Heading';
import UserMenu from '../UserMenu';
import * as S from './styles';
import user from 'data/constants/mockUsuario';

export default function Header() {
  return (
    <S.Wrapper>
      <Heading className="salutation" level={1} size="xxlarge">
        Ol√° {user.name.split(' ')[0]}
      </Heading>

      <UserMenu />
    </S.Wrapper>
  );
}
