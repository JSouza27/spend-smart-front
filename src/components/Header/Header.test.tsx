import { render, screen } from 'utils/test-utils';

import Header from '.';
import user from 'data/mocks/mockUsuario';

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = render(<Header />);

    expect(
      screen.getByRole('heading', { name: user.name })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
