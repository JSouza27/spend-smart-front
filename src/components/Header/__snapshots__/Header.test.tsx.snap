// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
.c1 {
  color: neutral_900;
  font-size: xxxlarge;
  font-weight: 600;
  line-height: 4rem;
}

.c4 {
  color: neutral_900;
  font-size: xlarge;
  font-weight: 600;
  line-height: 2.4rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 span {
  color: #474D66;
  line-height: 2.4rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #D8DAE5;
  border: none;
  border-radius: 100%;
  color: #696F8C;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5rem;
  overflow: hidden;
  width: 5rem;
}

.c0 {
  background-color: #F4F6FA;
  border: none;
  border-radius: 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 1.6rem;
  width: 100%;
}

.c0 .salutation {
  display: none;
}

@media screen (min-width:767px) {
  .c0 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }

  .c0 .salutation {
    display: block;
  }
}

<div
  class="c0"
>
  <h1
    class="c1 salutation"
  >
    Olá 
    João
  </h1>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        João da Silva
      </h2>
      <span>
        jjjjoao@xmail.com
      </span>
    </div>
    <div
      class="c5"
    >
      <h1
        class="c4"
      >
        JS
      </h1>
    </div>
  </div>
</div>
`;
